tickets/src/components/content/purchase/Ticket.js

```diff
@@ -22,29 +21,11 @@ export class Ticket extends Component {
     super(props)
 
     this.state = {
-      signed: false,
       email: '',
       sent: false,
     }
   }
 
-  componentDidUpdate = () => {
```

Comment:
We're not signing in the component anymore but from the middleware as soon as the key is "known". This makes rendering faster.